### PR TITLE
Add warning on tab close

### PR DESCRIPTION
**Pull request by JustusFT**
Ref: #80 

When the user tries to navigate away during a transfer, they will be prompted with a warning that the transfer will cancel. Note that newer browser versions will not show the custom error message but instead a generic warning. See this for more information: https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event#compatibility_notes

Closing the tab or clicking a link will trigger the warning on the following screens:
- ReceiveConsentScreen
- ReceiveProgressScreen
- SendInstructionsScreen
- SendProgressScreen

### CODE REVIEW

**Comment by JustusFT** [comment]
Also, maybe we should do something about this link to the terms in `ReceiveConsentScreen`, such as having it open in a new tab or in a modal to prevent the warning from happening.

Screen Shot 2022-12-07 at 07 53 44

Also, a better solution later on could be that the user is allowed to view other pages during a transfer, and have a way to return to the ongoing transfer screen

**Comment by donpui** [comment]
> Also, a better solution later on could be that the user is allowed to view other pages during a transfer, and have a way to return to the ongoing transfer screen

This would be nice thing. Probably we will need this when we will have dilation and multiple file transfer sending.

**Comment by donpui** [comment]
> Also, maybe we should do something about this link to the terms in `ReceiveConsentScreen`, such as having it open in a new tab or in a modal to prevent the warning from happening.

Probably this is the case, where we need to force user open in new tab.

**Comment by donpui** [comment]
FYI: On iOS Safari/Chrome this doesn't work.. however with Android Chrome it works.

**Comment by donpui** [comment]
`Copy link` button doesn't work for me. @JustusFT can you verify on your side?

**Comment by donpui** [comment]
Desktop Chrome/Firefox/Safari are working

**Comment by JustusFT** [comment]
@donpui Looks like the mobile clipboard issue was present for a while. I made a separate PR for the fix: https://github.com/LeastAuthority/winden/pull/112

**Comment by donpui** [comment]
> @donpui Looks like the mobile clipboard issue was present for a while. I made a separate PR for the fix: #112

This helped, thanks

File: client/src/app/components/screens/receive/ReceiveConsentScreen.tsx [hunk 1]
```diff
@@ -38,7 +38,14 @@ export function ReceiveConsentScreenContent(props: ContentProps) {
         </Button>
         <Text color="dark-grey" weight={400} size={14.4}>
           By using Winden you agree to the{" "}
-          <Anchor component={Link} to="/terms" color="tertiary" weight={600}>
+          <Anchor
+            component={Link}
+            to="/terms"
+            color="tertiary"
+            weight={600}
+            target="_blank"
```
Comment (donpui) [inline client/src/app/components/screens/receive/ReceiveConsentScreen.tsx:46]: This doesn't work. Still popups warning for leaving website. If I click, still opens in the same page.

**Pull request merged**